cdap-proto/src/main/java/io/cdap/cdap/proto/artifact/AppDeletionMessage.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.cdap.proto.artifact;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import io.cdap.cdap.proto.id.EntityId;
+
+/**
+ * A container for messages in the app deletion topic.
+ * It carries the payload as {@link JsonElement} which consists of the app-id.
+ */
+public final class AppDeletionMessage {
+    private final EntityId entityId;
+    private final JsonElement payload;
```

Comment:
Why do you need a JsonElement here? Is it supposed to be able to carry different classes?

Reply:
the payload is the app-id : it is parsed as a json. I am trying to maintain consistency with other subscribers in the codebase

Reply:
In other subscribers it's done to be able to pass objects of different classes in the same topic and decide the class dynamically with logic and not during deserialization. I don't see such a requirement here.

Reply:
should we just have Application-id instead?

Reply:
Yes. Just use whatever class you need to pass